lib/trento.pm

```diff
@@ -930,6 +931,70 @@ sub podman_self_check {
     assert_script_run('df -h');
 }
 
+=head3 podman_detect_status
+
+Check for status of running container with gived name.
+Polling state until container Status become Exit or timeout.
+Return the container Exit status. two possible `podman ps` output are:
+
+    Up 39 seconds,GIGIOM
+    Exited (0) 38 seconds ago,GIGIOM
+
+=over 2
+
+=item B<NAME> - String to filter the podman ps
+
+=item B<TIMEOUT> - Timeout waiting container to exit
+
+=back
+=cut
+
+sub podman_detect_status {
+    my (%args) = @_;
+    croak 'Missing mandatory name argument' unless $args{name};
+    croak 'Missing mandatory timeout argument' unless $args{timeout};
+
+    my $cmd = join(' ',
+        'podman', 'ps',
+        '--all',
+        '--filter', '"name=' . $args{name} . '"',
+        '--format', '"{{.Status}},{{.Names}}"');
+    my $start_time = time();
+    my $done = 0;
+    my $ret = undef;
+    while ((time() - $start_time <= $args{timeout}) && (!defined $ret)) {
+        # regexp here has to match --format from the previous podman command
+        $ret = $1 if (
```

Comment:
I have a doubt about this, why use $ret=$1, what represents $1, exit the loop?

Reply:
`$1` refer to regexp first group from next regexp. In particular it get value from this part `(\d+)`